other_processing/1.convert_sqlite_to_parquet/check.py

```diff
@@ -0,0 +1,105 @@
+import glob
+from hashlib import md5
+from pathlib import Path
+from typing import Optional, Union
+
+# credit: https://sebest.github.io/post/a-quick-md5sum-equivalent-in-python/
+def generate_md5_hash(
+    fpath: Union[str, Path], buffer_size: Optional[int] = 8192
+) -> tuple[str, str]:
+    """Creates a md5 hash of all contents within given file.
+
+    Parameters
+    ----------
+    fpath : Path
+        path to file to hash
+
+    buffer_size : int
+        chuck of data loaded into memory when hashing. The higher the value, the
+        more memory it uses. Default is 8192 bytes.
+    Returns
+    -------
+    str
+        md5 hash id that represents the file and it's contents
+
+    Raises
+    ------
+    FileNotFoundError
+        raised if invalid path or a non-file object is provided (E.G: directory)
+    ValueError
+        raised if provided file does not end with `.parquet`
+    """
+    # converting str into Path object
+    if isinstance(fpath, str):
+        fpath = Path(fpath)
+
+    # Checking
+    if not fpath.is_file():
+        e_msg = f"Provided file {str(fpath)} is not valid or not a file"
+        raise FileNotFoundError(e_msg)
+    if fpath.suffix != ".parquet":
+        raise ValueError("Path provided does not point to a parquet file")
+
+    # initializing hashing object
+    md5_obj = md5()
```

Comment:
With later versions of Python and hashlib, an argument `usedforsecurity` was added to help specify whether security-based use is occurring (see [Hashlib "Changed in version 3.9:..."](https://docs.python.org/3/library/hashlib.html#hash-algorithms)). Would the following be appropriate here based on the environment and context?
```suggestion
    md5_obj = md5(usedforsecurity=False)
```

Reply:
@d33bs I am not quite familiar with `usedforsecurity` after doing some reading, it seems that `usedforsecurity` is used to create a secured hash.

Do you think that applying a secured hashing algorithm to generate the hash will maintain reproducibility?

Reply:
@axiomcura - completely optional, I suggested this mostly for code readability and to avoid flagging potential security linting (certain libraries are very prone to false positives). From [Python's documentation](https://docs.python.org/3/library/hashlib.html):
> "usedforsecurity ... False indicates that the hashing algorithm is not used in a security context, e.g. as a non-cryptographic one-way compression function."

Reply:
added! Thanks for the explanation!